core/arch/arm/plat-sam/matrix.c

```diff
@@ -484,10 +484,6 @@ static int matrix_set_peri_security(unsigned int matrix, unsigned int peri_id)
 
 	bit = (0x01 << (peri_id % 32));
 
-	/* The Peripheral ID to SPSELR register bit mapping breaks at ID 73 */
```

Comment:
@clementleger 

Unless I am missing something, it looks like the datasheet still indicates the Peripheral ID to SPSELR mapping breaking the 1:1 mapping above 73 (i.e. needs to be offset by 1). I don't remember explicitly testing it, so if your testing indicated otherwise then might be nice to drop a line to Microchip to update the datasheet.
i.e. ID 74 maps to MATRIX_SPSELR3[10] and so on..
Sorry for the delayed response and happy to see all the nice improvements and substantial additions to the platform :)

https://ww1.microchip.com/downloads/en/DeviceDoc/DS60001476B.pdf
![image](https://user-images.githubusercontent.com/1905817/154114401-81eb431b-097c-4b43-9528-61c597cde83c.png)